tools/extra/hssi/hssiloopback.py

```diff
@@ -0,0 +1,164 @@
+# ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+# LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+# CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+# SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+# INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+# CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import re
+import os
+import glob
+import argparse
+import sys
+import traceback
+import fcntl
+import stat
+import struct
+import mmap
+from hssicommon import *
+
+
+class FPGAHSSILPBK(HSSICOMMON):
+    def __init__(self, args):
+        self.en = args.en
+        self.hssi_grps = args.hssi_grps
+        self.pcieaddress = args.pcieaddress
+        HSSICOMMON.__init__(self)
+
+    def hssi_loopback_en(self):
+        """
+        clear ctl address and ctl sts CSR
+        write 0x7/0x8 value ctl address
+        write 0x2 value ctl sts csr
+        poll for status
+        clear ctl address and ctl sts CSR
+        """
+        print("eth_group_loopback_en", self.en)
+
+        self.open(self.hssi_grps[0][0])
+
+        ctl_addr = hssi_ctl_addr(0)
+        if (self.en):
+            ctl_addr.sal_cmd(HSSI_SALCMD.ENABLE_LOOPBACK.value)
+        else:
+            ctl_addr.sal_cmd(HSSI_SALCMD.DISABLE_LOOPBACK.value)
+
+        cmd_sts = hssi_cmd_sts(0)
+        cmd_sts.value = 0x2
+
+        ret = self.clear_ctl_sts_reg(0)
+        if not ret:
+            print("Failed to clear HSSI CTL STS csr")
+            return 0
+
+        self.write32(0, HSSI_CSR.HSSI_CTL_ADDRESS.value, ctl_addr.value)
+        self.write32(0, HSSI_CSR.HSSI_CTL_STS.value, cmd_sts.value)
+
+        if not self.read_poll_timeout(0,
+                                      HSSI_CSR.HSSI_CTL_STS.value,
+                                      0x2):
+            print("HSSI ctl sts csr fails to update ACK")
+            return 0
+
+        ret = self.clear_ctl_sts_reg(0)
+        if not ret:
+            print("Failed to clear HSSI CTL STS csr")
+            return 0
+
+        return 0
+
+    def hssi_loopback_start(self):
+        """
+        print hssi info
+        enable/disable hssi loopback
+        """
+        print("----hssi_loopback_start----")
+        self.hssi_info(self.hssi_grps[0][0])
+        self.hssi_loopback_en()
+
+
+def main():
+    """
+    parse input arguemnts pciaddress and mtu
+    enum fpga pcie devices and find match
+    enable/disable loopback
+    """
+
+    parser = argparse.ArgumentParser()
+    pcieaddress_help = 'bdf of device to program \
+                       (e.g. 04:00.0 or 0000:04:00.0).' \
+                       ' Optional when one device in system.'
+    parser.add_argument('--pcieaddress', '-P',
```

Comment:
Could we use --pcie-address?

Reply:
changed input arguments to --pcie-address , fixed